### PR TITLE
Some tweaks to options

### PR DESCRIPTION
**Pull request by mtzguido**
Mostly making this to check if it sounds sane to you guys. It introduces 3 changes:

1- Fix short options. We had none, but I'd like to add `-f` to be `--force`

2- Introduce `--force` to skip the cache for files provided as CLI arguments. It happens often to me that I want to regenerate hints, quake test, or simply run again some file for which a valid `.checked` file exists. Instead of (finding it and) deleting it, this allows me to just pass `-f` (or `--force` :-) ) 

3- Introduce `--fuel min,max` and `--ifuel min,max`, simply to reduce verbosity.

### CODE REVIEW

File: src/basic/FStar.Options.fs [hunk 1]
```diff
@@ -778,6 +782,47 @@ let rec specs_with_types () : list<(char * string * opt_type * string)> =
...
+       ( noshort,
+        "fuel",
+        PostProcessed
+            ((function | String s ->
+                         let p f = Int (int_of_string f) in
+                         let min, max =
+                           match split s "," with
+                           | [f] -> f, f
+                           | [f1;f2] -> f1, f2
+                           | _ -> failwith "unexpected value for --fuel"
```
Comment (aseemr) [inline src/basic/FStar.Options.fs]: Can we change such `failwith`(s), that are not "Impossible!", to `raise_error` with `Errors.FailToProcessPragma`?

File: src/basic/FStar.Options.fs [hunk 2]
```diff
@@ -778,6 +782,47 @@ let rec specs_with_types () : list<(char * string * opt_type * string)> =
...
+       ( noshort,
+        "ifuel",
+        PostProcessed
+            ((function | String s ->
+                         let p f = Int (int_of_string f) in
+                         let min, max =
+                           match split s "," with
+                           | [f] -> f, f
+                           | [f1;f2] -> f1, f2
+                           | _ -> failwith "unexpected value for --fuel"
```
Comment (aseemr) [inline src/basic/FStar.Options.fs]: `fuel` -> `ifuel`?

**Comment by aseemr** [comment]
Thanks Guido, the changes look fine to me, added a couple of comments. However I am not convinced about adding a new `--force` option. There is already `--cache_off`, would it be better to have `--cache_off` take an (optional?) namespace string as argument to satisfy your usecase?

**Comment by s-zanella** [comment]
Thanks! The new `--fuel` and `--ifuel` options are very handy and having short options is a good idea.
I second Aseem's comment about `--force`. We already have `--cache_off` that could be extended with a pattern argument to subsume `--force`.

File: src/basic/FStar.Options.fs (hunk 2 above)
Comment (mtzguido) [reply src/basic/FStar.Options.fs]: oops thanks!

File: src/basic/FStar.Options.fs (hunk 1 above)
Comment (mtzguido) [reply src/basic/FStar.Options.fs]: These are actually caught by our getopt and we get an error like this:
```
$ ./bin/fstar.exe --fuel 1,1,1 ulib/prims.fst 
wrong argument given to option `fuel`
```
So I guess raising anything is equivalent... I could rework this getopt logic I guess (and maybe use something more standard?)

Comment (aseemr) [reply src/basic/FStar.Options.fs]: This is fine too, as long as in the emacs mode the cursor does not go to the beginning of the file and says "F* may be in an inconsistent state" (which is what happens with `failwith` normally I think?).

**Comment by mtzguido** [comment]
Thanks for the comments guys.

I would find a namespace argument to --cache_off a lot less convenient, and would probably just delete the checked files then. I wanted `-f` since it's really just two keystrokes, but I don't mind so much. Should I revert that commit and merge the rest?

**Comment by aseemr** [comment]
That sounds good to me, thanks!

**Comment by nikswamy** [comment]
What about:

* `-f` as force for the command line arguments.
* `--force <namespace selector>'
* and --cache_off as `--force '*'` (or removing --cache_off)

?

**Comment by mtzguido** [comment]
Just realized that the shuffling code in the benchmark rule for ulib

https://github.com/FStarLang/FStar/blob/9a53600af7c577d463f6f5a7c3acccf8a08be7cd/ulib/Makefile.verify#L62-L72

Also prevents running the benchmark reliably in parallel, since some invocations will have missing .checked files for their dependencies and do extra work that we don't want to measure. So I'd like a `-f` to avoid this, preferably one that doesn't take a namespace (at the surface).

I would just add `-f` and keep `--cache_off` as this. I don't think we'll have a need for selectively disabling the cache, but I could implement a namespace argument `--force` and do as Nik suggested.

**Pull request merged**